packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -1626,21 +1638,26 @@ export class Bucket extends BucketBase {
   public static fromBucketAttributes(scope: Construct, id: string, attrs: BucketAttributes): IBucket {
     const stack = Stack.of(scope);
     const region = attrs.region ?? stack.region;
-    const urlSuffix = stack.urlSuffix;
+    const regionInfo = regionInformation.RegionInfo.get(region);
+    const urlSuffix = regionInfo.domainSuffix ?? stack.urlSuffix;
 
     const bucketName = parseBucketName(scope, attrs);
     if (!bucketName) {
       throw new Error('Bucket name is required');
     }
     Bucket.validateBucketName(bucketName);
 
-    const newUrlFormat = attrs.bucketWebsiteNewUrlFormat === undefined
-      ? false
-      : attrs.bucketWebsiteNewUrlFormat;
+    const oldEndpoint = `s3-website-${region}.${urlSuffix}`;
+    const newEndpoint = `s3-website.${region}.${urlSuffix}`;
+
+    let staticDomainEndpoint = regionInfo.s3StaticWebsiteEndpoint ?? oldEndpoint;
```

Comment:
I think I would like us to use

https://github.com/aws/aws-cdk/blob/92dc4372e4e86d27a276cfce9c832b71b9291d8b/packages/%40aws-cdk/core/lib/stack.ts#L1072

If `bucketWebsiteNewUrlFormat` is not set. That will get us the right value for the endpoint even if the region is  unresolvable. The default should be the new format.

If `bucketWebsiteNewUrlFormat` is set, we can pick old or new behavior specifically.

Reply:
We still have to use `regionInfo.s3StaticWebsiteEndpoint` to give priority to an explicitly provided region. Good fallback for fully agnostic environments though!